File: insurance-management/src/main/java/com/insurance/insurancemanagement/responseEntity/ApiResponse.java
package com.insurance.insurancemanagement.responseEntity;

import java.util.List;

import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.Setter;

@Getter @Setter @AllArgsConstructor
public class ApiResponse {

  private ApiResponseStatus status;

  private Object data;

  private List<ApiError> errors;

  private List<ApiWarning> warnings;

  public ApiResponse() {
    super();
  }

  public ApiResponse(ApiResponseStatus status, Object data) {
    super();
    this.status = status;
    this.data = data;
  }

  public ApiResponse(ApiResponseStatus status, Object data,
      List<ApiError> errors) {
    super();
    this.status = status;
    this.data = data;
    this.errors = errors;
  }

}

File: insurance-client/src/app/models/api-response.ts
import {ApiResponseStatus} from './api-response-status.enum'
import {ApiError} from './api-error'
import {ApiWarning} from './api-warning'

export interface ApiResponse {
    status : ApiResponseStatus;
    data : any;
    errors : Array<ApiError>;
    warnings : Array<ApiWarning>
}
